pyuvdata/uvbeam.py

```diff
@@ -436,6 +442,80 @@ def set_phased_array(self):
         self._gain_array.required = True
         self._coupling_matrix.required = True
 
+    def efield_to_power(self, calc_cross_pols=True, keep_basis_vector=False,
+                        run_check=True, check_extra=True, run_check_acceptability=True):
+        """
+        Convert E-field beam to power beam.
+
+        Args:
+            calc_cross_pols: If True, calculate the crossed polarization beams
+                (e.g. 'xy' and 'yx'), otherwise only calculate the same
+                polarization beams (e.g. 'xx' and 'yy'). Default is True.
+            keep_basis_vector: If True, keep the directionality information and
+                just multiply the efields for each basis vector separately
+                (caution: this is not what is standardly meant by the power beam).
+                Default is False.
+            run_check: Option to check for the existence and proper shapes of
+                required parameters after converting to power. Default is True.
+            check_extra: Option to check optional parameters as well as
+                required ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                required parameters after combining objects. Default is True.
+        """
+        if self.beam_type is not 'efield':
+            raise ValueError('beam_type must be efield')
+
+        efield_data = self.data_array
+        efield_naxes_vec = self.Naxes_vec
+
+        feed_pol_order = [(0, 0)]
+        if self.Nfeeds > 1:
+            feed_pol_order.append((1, 1))
+
+        if calc_cross_pols:
+            self.Npols = self.Nfeeds ** 2
+            if self.Nfeeds > 1:
+                feed_pol_order.extend([(0, 1), (1, 0)])
+        else:
+            self.Npols = self.Nfeeds
+
+        pol_strings = []
+        for pair in feed_pol_order:
+            pol_strings.append(self.feed_array[pair[0]] + self.feed_array[pair[1]])
+        self.polarization_array = np.array([uvutils.polstr2num(ps.upper()) for ps in pol_strings])
+
+        if not keep_basis_vector:
+            self.Naxes_vec = 1
+
+        # adjust requirements, fix data_array form
+        self.set_power()
```

Comment:
yes, thanks, I'll fix that.

Reply:
Actually, they are called in cst_beam.py and beamfits.py because they are used when reading in beams. Those are sort of internal calls, but they are different files... I could go either way, but our other objects have similar set routines that don't have an underscore. Maybe we should make all of them underscores, but I think we should do that in a separate PR.

Reply:
Ok, no problem. I don't have a preference for how it's done, I was just thinking we should be consistent. But it sounds like this is the consistent way, so I'm happy.